### PR TITLE
Add nf tests for the picard bam qc subworkflow

### LINKED ISSUE #4473: [FEATURE] Add nf-test to subworkflow bam_qc_picard
### Is your feature request related to a problem? Please describe

Migration of tests from py-tests to nf-tests

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by nikhil**
## PR checklist

Closes #4473 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: subworkflows/nf-core/bam_qc_picard/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                ]
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert workflow.success},
+                { assert snapshot(
+                    workflow.out.coverage_metrics[0].get(0),
```
Comment (Puumanamana) [inline subworkflows/nf-core/bam_qc_picard/tests/main.nf.test:46]: For consistency, I'd only use the bracket notation or the .get() notation (unless there's a reason to use different ones here?)
e.g., `workflow.out.coverage_metrics[0][0]` or `workflow.out.coverage_metrics.get(0).get(0)`

File: tests/config/pytest_modules.yml
Comment (Puumanamana) [inline tests/config/pytest_modules.yml:1]: you probably need to remove those lines from this file, right? I believe that's one of the reason the linting fails

**Review by Puumanamana — COMMENTED** [review]
Looks good! Added a few suggestions

File: tests/config/pytest_modules.yml [hunk 2]
```diff
@@ -3432,6 +3432,10 @@ subworkflows/bam_ngscheckmate:
   - subworkflows/nf-core/bam_ngscheckmate/**
   - tests/subworkflows/nf-core/bam_ngscheckmate/**
 
+subworkflows/bam_rseqc:
+  - subworkflows/nf-core/bam_rseqc/**
+  - tests/subworkflows/nf-core/bam_rseqc/**
+
 subworkflows/bam_qc_picard:
```
[the remark this answers is not in the document]
Comment (nikhil) [reply tests/config/pytest_modules.yml]: Good catch! It should be resolved now

File: subworkflows/nf-core/bam_qc_picard/tests/main.nf.test (hunk 1 above)
Comment (nikhil) [reply subworkflows/nf-core/bam_qc_picard/tests/main.nf.test]: Good point! It should be done now

File: tests/config/pytest_modules.yml
Comment (nikhil) [reply tests/config/pytest_modules.yml:1]: Done!

**Review by SPPearce — APPROVED** [review]
Looks good to me.

**Pull request merged**